src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/common.kt

```diff
@@ -159,6 +168,100 @@ fun updateTransactionWithRefundEvent(
     .thenReturn(transaction)
 }
 
+fun handleGetState(
+  tx: BaseTransaction,
+  event: TransactionAuthorizationRequestedEvent,
+  npgStateService: NpgStateService,
+  transactionsServiceClient: TransactionsServiceClient,
+  tracingInfo: TracingInfo,
+  retryCount: Int = 0
+): Mono<BaseTransaction> {
+  return Mono.just(tx)
+    .cast(BaseTransactionWithRequestedAuthorization::class.java)
+    .filter { transactions ->
+      transactions.transactionAuthorizationRequestData.paymentGateway ==
+        TransactionAuthorizationRequestData.PaymentGateway.NPG
+    }
```

Comment:
here I would add a switchIfEmpty returning a mono error with an exception that will trace the fact that an authorization requested event has been received for an unexpected payment gateway (such as REDIRECT)